multiuser/api/che-multiuser-api-permission/src/main/java/org/eclipse/che/multiuser/api/permission/server/InstanceParameterValidator.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.multiuser.api.permission.server;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.eclipse.che.api.core.BadRequestException;
+import org.eclipse.che.api.core.NotFoundException;
+import org.eclipse.che.commons.annotation.Nullable;
+
+/**
+ * Validates that provided instance parameter is valid
+ *
+ * @author Sergii Leschenko
+ */
+@Singleton
+public class InstanceParameterValidator {
+  private final PermissionsManager permissionsManager;
+
+  @Inject
+  public InstanceParameterValidator(PermissionsManager permissionsManager) {
+    this.permissionsManager = permissionsManager;
+  }
+
+  /**
+   * Validates that provided instance parameter is valid for specified domain
+   *
+   * @param domain the domain of specified {@code instance}
+   * @param instance the instance to check
+   * @throws BadRequestException if specified {@code domain} is null
+   * @throws BadRequestException if specified {@code instance} is not valid
+   * @throws NotFoundException if specified {@code domain} is unsupported
+   */
+  public void validate(String domain, @Nullable String instance)
+      throws BadRequestException, NotFoundException {
+    checkArgument(domain != null, "Domain id required");
+    if (permissionsManager.getDomain(domain).isInstanceRequired() && instance == null) {
+      throw new BadRequestException("Specified domain requires non nullable value for instance");
+    }
+  }
```

Comment:
@sleshchenko proposal:
`
  checkArgument(domain != null, "Domain id required");
    checkArgument((permissionsManager.getDomain(domain).isInstanceRequired() && instance == null),
        "Specified domain requires non nullable value for instance"
    );
`

Reply:
It can be rewritten with `checkArgument` method. But then condition should be rewritten as
```
checkArgument(!permissionsManager.getDomain(domain).isInstanceRequired() || instance != null), "Specified domain requires non nullable value for instance");
```
So I'd not change it because then condition becomes more complicated.

Reply:
We can remove negation in the `checkArgument` so than condition will be the same (and `domain == null` in first check )

Reply:
This method is named after Precondition#checkArgument from guava. And I think it is
logical to have negation there while it is named in this way. Up to you change it or not =)